### DISCUSSION TITLE
Apple 27" Thunderbolt Display permanently dimmed?!

### DISCUSSION THREAD

**Discussion by azxiety** [MonitorControl/MonitorControl#970] [Q&A]
Hi guys, I was curious if I could dim the Apple Thunderbolt display and saw a video on YouTube with a guy using MonitorControl to adjust his other monitors. I didn't watch the whole video and just assumed the Thunderbolt Display would be no different. 

Upon installing the application my Display immediately went dim and that was it's new max brightness. 
I've tried to manually set the DDC Max to 250 / 255 and nothing. 

I'm currently running a MacBook Pro 16" M1 Pro with MacOS Monterey 12.2.1 

I've tried using the method on https://discussions.apple.com/thread/252063309?answerId=255777656022#255777656022 which I got from a post from someone in this group - https://github.com/MonitorControl/MonitorControl/discussions/739

Any advice and or fixes would be much appreciated!

**Comment by waydabber** [comment]
Hi @azxiety - you might want to try ddcct (on Intel) or m1ddc to try to set the brightness to 255 - hopefully that works. Also you can try Lunar.fyi as that app has a reset option that is reported to work on the 27" Thunderbolt Display to reset brightness to default. It is not clear why MonitorControl has issues with some older Apple displays using DDC and not with others - a developer with a hands-on with such a display would be needed to experiment with this a little. It might be something simple like modifying DDC command timings, repeating commands, or issue commands in a different fashion. DDC is not officially supported by these displays and it looks like hardware and firmware revision differences might cause some units to work and some less-so.

@alin23

**Comment by HalfWhitt** [comment]
I encountered the same issue (with the 27" Cinema Display, which is almost the same monitor). I can fix it by connecting the monitor to my older Intel Macbook and controlling the brightness from there, or by using Lunar instead of using MonitorControl. I'm afraid I've never messed with Swift or with any hardware control programming, or I'd try to help out with looking into it.

**Comment by trosicki** [comment]
I have a new MacBook Pro M1 and could not get my 27" Cinema display to get beyond 67% brightness.  Lunar.fyi fixed it.  It now goes to 100% now which is likely the default 100% brightness of the monitor originally. Thanks, waydabber!  Now, if I wanted it to go beyond 100% is that also doable?  Thanks

**Reply by waydabber** [reply]
Hi @trosicki - how did you make it go back to 100% exactly in Lunar? I did talk to @alin23 (the dev of Lunar) about it and he has no idea why and how Lunar can supposedly fix Cinema Displays messed up by DDC commands and he is getting the same complaints about Lunar making some of these displays dim - yet it is obvious that some users (like you) somehow are able to restore their displays with Lunar. :)

**Reply by SoCalBeachrunner** [reply]
@ the developers of MonitorControl and Lunar... perhaps some clue about the inconsistent behavior with Apple LED Cinema displays... Just chiming in that I had a similar/same problem using 2021 16" MacBook Pro M1 and 24" LED Cinema Display A1267 where the brightness abruptly dropped to (I'm guessing) 67% as the MonitorControl installation completed and was stuck there with MonitorControl thinking that was "the new" 100%. Neither power cycling the Cinema display nor rebooting helped.  Simply installing Lunar.fyi fixed it. Something I noticed that might offer some clue as to why it works for some and not others is that I happened to install MonitorControl in a windowed room as the natural light was getting dim. When installing Lunar I had by then turned the ceiling lights on. Maybe something in one installation or the other is detecting/reading/messing with values influenced by an ambient light reading during the installation / initial calibration process that runs automatically during first time setup. Just throwing a bone and a hunch.

**Reply by SoCalBeachrunner** [reply]
+ when installing MonitorControll, the Cinema display's USB connector was not connected to the MBP. By the time I ran the Lunar installer it was connected using a generic USB A -> USB C adapter. Since with older Macs the A1267 requires the USB connection to control the monitor, maybe that difference comes into play somehow, too, regarding difference outcomes for some users. Straws... feel free to grab one LOL

**Reply by alin23** [reply]
> when installing MonitorControll, the Cinema display's USB connector was not connected to the MBP

I think this is the only factor here. 

On some of these displays `255` is the highest brightness value, while on others it is still `100`.

When the **USB is connected**, both MonitorControl and Lunar will prefer using the Apple Native protocol which bypasses DDC and that `100/255` limit. That will allow the brightness to go to real maximum **if the system also allows it**.

![image](https://user-images.githubusercontent.com/3819725/207578325-2bc503fd-f0f6-46ee-a4a7-ccd032ae85e2.png)

When macOS has that bug where it's blocked at a lower than max brightness, **disconnecting the USB** will make Lunar and MonitorControl use DDC instead and set the brightness to 255, unlocking that over-100 range of values.

![image](https://user-images.githubusercontent.com/3819725/207578369-00aafda5-ed5c-49f4-992f-3699cee94e1a.png)


So it might just be a matter of connecting/disconnecting the USB cable, and toggling between 255 and 100 max DDC brightness.

Ambient light definitely doesn't have any effect on what Lunar and MonitorControl are doing. It might have an effect on what macOS does, that I don't know.

**Comment by trosicki** [comment]
Originally, Lunar said "not detecting" but then after I kept proceeding to next next next...it finally took and seems to be working. My 27" Cinema display is now at 100% rather than stuck on 67% and I can control it through lunar.

**Reply by waydabber** [reply]
Does the backlight actually change with Lunar via DDC or is it software control (like when you use MonitorControl but disable DDC)?

**Reply by trosicki** [reply]
I am not technical enough to answer that question fully
 but I tried MonitorControl and all it did was show a sliding monitor brightness switch on my screen but it would brighten only to about halfway on the slide and then nothing further.  The rest of the way from like 60% to 100% didn't change at all.  After I changed to Lumen that unleashed the remaining 40% of brightness up to 100%.

**Reply by waydabber** [reply]
You'd need to untick the following setting in MonitorControl to fix this:

<img width="842" alt="Screen Shot 2022-04-21 at 20 39 50" src="https://user-images.githubusercontent.com/37590873/164531091-dacd547b-0b0d-4bb0-8ed1-716454120fc8.png">
 
technical stuff: MonitorControl uses software dimming by default between 0-50% and hardware from 50-100%. The latter does not work with your display so that's why brightness is not changing beyond 50% unless you untick that checkbox.

**Comment by luco** [comment]
I'm stuck in 50% as well. Tried MonitorControl and Lunar and it didn't worked.

**Reply by alin23** [reply]
If you still have Lunar installed, you could try the following Terminal command to check if it goes to the highest brightness:

```sh
/Applications/Lunar.app/Contents/MacOS/Lunar @ ddc all BRIGHTNESS 254
```

**Reply by luco** [reply]
Didn't work :/

```
➜  ~ /Applications/Lunar.app/Contents/MacOS/Lunar @ ddc all BRIGHTNESS 254
LED Cinema Display [ID 69511592]: Writing 254 for BRIGHTNESS: Ok
```

**Reply by luco** [reply]
MonitorControl also does not let me set DDC Max

**Reply by waydabber** [reply]
I think the issue is that MC detects the display as an Apple display that has native brightness control. If this happens with the latest Monterey, there might have been a change.

You can try [BetterDisplay](https://betterdisplay.pro) as that app has a specific option to treat Apple displays as third party so you can enable DDC. You can also change DDC brightness levels under the display. I Think Lunar has the same. If neither works, then I have no idea how to fix this. :)

**Reply by waydabber** [reply]
https://github.com/MonitorControl/MonitorControl/issues/1244

**Reply by luco** [reply]
I fixed it by... turning it off and on. lol
